src/Security/SslContext.cs

```diff
@@ -523,7 +523,7 @@ public SslStatus SetEncryptionCertificate (SecIdentity identify, IEnumerable<Sec
 			get {
 				IntPtr value;
 				result = SSLCopyPeerTrust (Handle, out value);
-				return (value == IntPtr.Zero) ? null : new SecTrust (value);
+				return (value == IntPtr.Zero) ? null : new SecTrust (value, true);
```

Comment:
Look at `SSLCopyPeerTrust()` - you must release the returned object when you're done with it.